src/util/TokenStreamBase.js

```diff
@@ -16,7 +16,7 @@ function TokenStreamBase(input, tokenData){
      * @property _reader
      * @private
      */
-    this._reader = input ? new StringReader(input.toString()) : null;
+    this._reader = input !== undefined ? new StringReader(input.toString()) : null;
```

Comment:
During the unit tests: `TokenStreamBase` is call with `undefined`. Maybe the parser need to know if the input is invalid (`undefined`) or valid (*String* / *StringReader*). So I keep `null` value in `this._reader`.

If invalid input and empty string are equivalent for the parser : the PR #186 is better.